data/bookkeeping/block_test.go

```diff
@@ -461,3 +461,76 @@ func TestInitialRewardsRateCalculation(t *testing.T) {
 	consensusParams.InitialRewardsRateCalculation = true
 	require.True(t, runTest())
 }
+
+func performRewardsRateCalculation(
+	t *testing.T, consensusParams config.ConsensusParams,
+	rewardsRate uint64,
+	rewardsLevel uint64, rewardsResidue uint64, rewardsRecalculationRound basics.Round,
+	incentivePoolBalance uint64, totalRewardUnits uint64, startingRound basics.Round) bool {
+	require.GreaterOrEqual(t, incentivePoolBalance, consensusParams.MinBalance)
+
+	curRewardsState := RewardsState{
+		RewardsLevel:              rewardsLevel,
+		RewardsResidue:            rewardsResidue,
+		RewardsRecalculationRound: rewardsRecalculationRound,
+		RewardsRate:               rewardsRate,
+	}
+
+	for rnd := int(startingRound); rnd < int(uint64(startingRound)+consensusParams.RewardsRateRefreshInterval+2); rnd++ {
+		nextRewardState := curRewardsState.NextRewardsState(basics.Round(rnd), consensusParams, basics.MicroAlgos{Raw: incentivePoolBalance}, totalRewardUnits)
+		// adjust the incentive pool balance
+		var ot basics.OverflowTracker
+
+		// get number of rewards per unit
+		rewardsPerUnit := ot.Sub(nextRewardState.RewardsLevel, curRewardsState.RewardsLevel)
+		require.False(t, ot.Overflowed)
+
+		// subtract the total dispersed funds from the pool balance
+		incentivePoolBalance = ot.Sub(incentivePoolBalance, ot.Mul(totalRewardUnits, rewardsPerUnit))
+		require.False(t, ot.Overflowed)
+
+		// make sure the pool retain at least the min balance
+		ot.Sub(incentivePoolBalance, consensusParams.MinBalance)
```

Comment:
why not use simple comparison?

Reply:
I think this does internally use that.

Reply:
It would just be more readable to do `require.GreaterOrEqual(incentivePoolBalance, consensusParams.MinBalance)`.